Add `KorUnSmile` task

ingyuseong opened the pull request:
## Add `KorUnSmile` task
Sanity check results are as follows.
- [KorUnSmile Dataset](https://github.com/smilegate-ai/korean_unsmile_dataset)
- Data sample (⚠️ As it is dataset about hate speech, following examples might include offensive expressions or swears.)
  ```
  !!@@##@@!! -- Example 1
  엌 틀무새들 좆됐노 ㅋㅋㅋㅋ 혐오

  틀딱잘못은 맞는데 점주도 싸가지 없이 얘기했겠네 손모양만 봐도 딱 싸가지없게 얘끼했네 혐오

  응안뽑앗어 양호

  우리나라 페미들은 이슬람/인도의 여성대우가 우리보다 좋다고 씨부린다그런소리를 하는거 부터가 그냥 미친거야 혐오

  반대박고 베댓들접어라익이야노 악플

  여자들은 취미가 애낳는건가.. 취미를 좀 가져라
  ```
- Passed unit test
- Few-shot performance by `polyglot-ko-1.3b`
  - \# shots: 0                                                                                                                                                                                             
    gpt2 (pretrained=EleutherAI/polyglot-ko-1.3b), limit: None, provide_description: False, num_fewshot: 0, batch_size: 4
    |   Task   |Version|Metric|Value |   |Stderr|
    |----------|------:|------|-----:|---|-----:|
    |korunsmile|      0|f1    |0.3954|±  | 0.008|
  - \# shots: 1
    gpt2 (pretrained=EleutherAI/polyglot-ko-1.3b), limit: None, provide_description: False, num_fewshot: 1, batch_size: 4
    |   Task   |Version|Metric|Value |   |Stderr|
    |----------|------:|------|-----:|---|-----:|
    |korunsmile|      0|f1    |0.3618|±  | 0.008|
  - \# shots: 10
    gpt2 (pretrained=EleutherAI/polyglot-ko-1.3b), limit: None, provide_description: False, num_fewshot: 10, batch_size: 4
    |   Task   |Version|Metric|Value |   |Stderr|
    |----------|------:|------|-----:|---|-----:|
    |korunsmile|      0|f1    |0.3158|±  |0.0069|

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=430) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=430) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=430) it.</sub>

ingyuseong commented on lm_eval/tasks/korunsmile.py:
```diff
@@ -0,0 +1,71 @@
+
+    def validation_docs(self):
+        return map(self._process_doc,self.dataset["valid"])
+
```
@cardy20 Because `construct_requests` is inherited from `MultipleChoiceTask`, there's no need to define `construct_requests` again. In addition, I modified the overall setting to multi-class classification task that we discussed on previous meeting. Thank you for the helpful review :)

ingyuseong commented:
@cardy20 Because `construct_requests` is inherited from `MultipleChoiceTask`, there's no need to define `construct_requests` again. In addition, I modified the overall setting to multi-class classification task as we discussed on previous meeting. Thank you for the helpful review :)

StellaAthena commented:
Are there comparisons with established uses of this evaluation benchmark that we can use to confirm the implementation is working as expected?

ingyuseong commented:
@StellaAthena Unfortunately, there are no references :(

Because we simplify the evaluation setting—multi-label to multi-class classification—for the following reasons.

1. The original evaluation setting of the benchmark(multi-label) was not suitable for few-shot evaluation of autoregressive LMs. At least, it was too difficult to report meaningful score. It consists of two sub-tasks: 1) Multi-class classification among three classes. (`혐오표현`, `악플/욕설`, and `clean`) 2) Multi-label classification for seven different labels for `혐오표현`(only if the input is classified as `혐오표현`). So we discussed about simplification of the evaluation in the previous meeting. (Polyglot Korean team)

2. Simplified setting was enough to achieve our goal, as we aim to evaluate the model's ability to detect toxic text in few-shot setting, in a manner similar to Schick et al. (2021), on Korean Unsmile Dataset. So there is no reasons to follow the original complex setting.

As a result, we simplified the evaluation setting only to include sub-task #1. And that's why there are no reference scores to provide.
